lighthouse-core/test/scripts/run-mocha-tests.js

```diff
@@ -0,0 +1,322 @@
+      // test contamination by chance of mocha splitting up the work in a way that hides it.
+      default: Boolean(process.env.CI),
+    },
+    'bail': {
+      alias: 'b',
+      type: 'boolean',
+      default: false,
+    },
+    't': {
+      type: 'string',
+      describe: 'an alias for --grep, to run only tests with matching titles',
+    },
+    'onlyFailures': {
+      type: 'boolean',
+    },
+  })
+  .wrap(y.terminalWidth())
+  .argv;
+const argv =
+  /** @type {Awaited<typeof rawArgv> & CamelCasify<Awaited<typeof rawArgv>>} */ (rawArgv);
+
+// This captures all of our mocha tests except for:
+// * flow-report, because it needs to provide additional mocha flags
+// * various *-test-pptr.js integration tests, which are long so are handled explicitly in
+//   specific package.json scripts
+const defaultTestMatches = [
+  'build/**/*-test.js',
+  'clients/test/**/*-test.js',
+  'lighthouse-cli/**/*-test.js',
+  'lighthouse-core/**/*-test.js',
+  'lighthouse-core/test/fraggle-rock/**/*-test-pptr.js',
+  'report/**/*-test.js',
+  'shared/**/*-test.js',
+  'third-party/**/*-test.js',
+  'treemap/**/*-test.js',
+  'viewer/**/*-test.js',
+];
+
+const mochaPassThruArgs = argv._.filter(arg => typeof arg !== 'string' || arg.startsWith('--'));
+const filterFilePatterns = argv._.filter(arg => !(typeof arg !== 'string' || arg.startsWith('--')));
+
+function getTestFiles() {
+  // Collect all the possible test files, based off the provided testMatch glob pattern
+  // or the default patterns defined above.
+  const testsGlob = argv.testMatch || `{${defaultTestMatches.join(',')}}`;
+  const allTestFiles = glob.sync(testsGlob, {cwd: LH_ROOT, absolute: true});
+
+  // If provided, filter the test files using a basic string includes on the absolute path of
+  // each test file. Map back to a relative path because it's nice to keep the printed commands shorter.
+  // Why pass `absolute: true` to glob above? So that this works:
+  //     yarn mocha /Users/cjamcl/src/lighthouse/lighthouse-core/test/runner-test.js
+  let filteredTests = (
+    filterFilePatterns.length ?
+      allTestFiles.filter((file) => filterFilePatterns.some(pattern => file.includes(pattern))) :
+      allTestFiles
+  ).map(testPath => path.relative(process.cwd(), testPath));
+
+  if (argv.onlyFailures) {
+    const failedTests = getFailedTests();
+    if (failedTests.length === 0) throw new Error('no tests failed');
+
+    const titles = getFailedTests().map(failed => failed.title);
+    baseArgs.push(`--grep="${titles.map(escapeRegex).join('|')}"`);
+
+    filteredTests = filteredTests.filter(file => failedTests.some(failed => failed.file === file));
+  }
+
+  if (filterFilePatterns.length) {
+    console.log(`applied test filters: ${JSON.stringify(filterFilePatterns, null, 2)}`);
+  }
+  console.log(`running ${filteredTests.length} test files`);
+
+  return filteredTests;
+}
+
+const baseArgs = [
+  '--require=lighthouse-core/test/test-env/mocha-setup.js',
```

Comment:
what about a [.mocharc.js file](https://github.com/mochajs/mocha/blob/master/example/config/.mocharc.js)?

is going full CLI flag just easier given the permutations?

Reply:
That file just extracts what you'd pass on the CLI to a file. It doesn't do what jest was doing, which was setting up its own environment for just for particular tests. That's the entire thing run-mocha-tests.js is doing. so in short:

> is going full CLI flag just easier given the permutations?

Yeah